cql3/Cql.g

```diff
@@ -501,10 +502,10 @@ insertStatement returns [std::unique_ptr<raw::modification_statement> expr]
     @init {
         auto attrs = std::make_unique<cql3::attributes::raw>();
         std::vector<::shared_ptr<cql3::column_identifier::raw>> column_names;
-        std::vector<::shared_ptr<cql3::term::raw>> values;
+        std::vector<cql3::expr::expression> values;
         bool if_not_exists = false;
         bool default_unset = false;
-        ::shared_ptr<cql3::term::raw> json_value;
+        std::optional<cql3::expr::expression> json_value;
```

Comment:
No, we can also end up in the non-json variation.

Reply:
The non-json validation doesn't touch this variable.  The only code reading the variable dereferences it unconditionally.

Reply:
Right, and therefore I think it's correct to only initialize it if we ever use it. If I remove std::optional<>, then it only compiles because we happen to have a default constructor for expressions.

Reply:
OK.